examples/react-server/src/features/unocss/plugin.ts

```diff
@@ -12,112 +17,110 @@ import { createVirtualPlugin } from "../utils/plugin";
 // reading `uno.config.ts` is adding more than 1 sec on startup time. is it normal?
 
 export function vitePluginSharedUnocss(): PluginOption {
-  // reuse original plugin to grab internal unocss instance and transform plugins
-  const originalPlugins = vitePluginUnocss();
-  const apiPlugin = originalPlugins.find((p) => p.name === "unocss:api");
-  tinyassert(apiPlugin);
-  const ctx = (apiPlugin.api as UnocssVitePluginAPI).getContext();
-
-  // reuse unocss transform plugins with sharedDuringBuild
-  const transformPlugins = originalPlugins
-    .filter((plugin) => plugin.name.startsWith("unocss:transformers:"))
+  // reuse original plugins with sharedDuringBuild
+  const originalPlugins = vitePluginUnocss()
+    .filter(
+      (plugin) =>
+        plugin.name === "unocss:api" ||
+        plugin.name.startsWith("unocss:transformers:"),
+    )
     .map((plugin) => ({
       ...plugin,
       sharedDuringBuild: true,
     }));
 
-  const mainPlugin: Plugin = {
-    name: vitePluginSharedUnocss.name,
-    sharedDuringBuild: true,
+  const environmentPlugin: BoundedPluginConstructor = (environment) => {
+    const apiPlugin = environment.config.plugins.find(
+      (plugin) => plugin.name === "unocss:api",
+    );
+    tinyassert(apiPlugin);
+    const ctx = (apiPlugin.api as UnocssVitePluginAPI).getContext();
```

Comment:
Per-environment plugin callback doesn't share the same scope, so we need to grab unocss instance via shared plugin api.

Reply:
In alpha.10, `sharedDuringBuild` can be assigned to `BoundedPluginConstructor` as well https://github.com/hi-ogawa/vite-environment-examples/pull/64/commits/94cb16d1b91a9e94b3f5d17a129a1bbfb75d65c1